common/models/facility.js

```diff
@@ -496,20 +496,21 @@ module.exports = (Facility) => {
     ]
   });
 
-  Facility.facilitiesByService = async (service_name, district_id, cb) => {
-    if (!service_name){
-      return [];
-    }
+  Facility.facilitiesByService = async (service_name, districts, cb) => {
+    const IDs = await getDistrictsIDs(districts);
+    if (!service_name) return [];
+
     const serviceIds = await server.models.Service.find().filter(service => {
       return _.lowerCase(service.service_name).includes(_.lowerCase(service_name));
     }).map(service => service.id);
+
     const facilityIds = await server.models.FacilityService.find().filter(facilityService => {
       return serviceIds.includes(facilityService.service_id);
     }).map(facilityService => facilityService.facility_id);
-    const facilities = await server.models.Facility.find({where: {id: {inq: facilityIds}}});
-    if(district_id){
-      return [];
-    }
+
+    const where =  { id: {inq: facilityIds}};
+    if (IDs.length) where.district_id = { inq: IDs }
+    const facilities = await server.models.Facility.find({where: where});
```

Comment:
you can use the shorthand syntax here, i.e. `server.models.Facility.find({where})`